nautilus_trader/data/aggregation.pyx

```diff
@@ -734,23 +755,58 @@ cdef class TimeBarAggregator(BarAggregator):
         self._stored_open_ns = dt_to_unix_nanos(self.get_start_time(now))
         self._stored_close_ns = 0
         self._cached_update = None
-        self._build_with_no_updates = build_with_no_updates
-        self._timestamp_on_close = timestamp_on_close
-        self._composite_bar_build_delay = composite_bar_build_delay
         self._add_delay = bar_type.is_composite() and bar_type.composite().is_internally_aggregated()
         self._batch_open_ns = 0
         self._batch_next_close_ns = 0
-        self._time_bars_origin = time_bars_origin
-        self._skip_first_non_full_bar = skip_first_non_full_bar
 
-        if interval_type == "left-open":
-            self._is_left_open = True
-        elif interval_type == "right-open":
-            self._is_left_open = False
+
+
+    def _validate_time_bars_origin(self):
+        # TODO: Tests for DateOffset time_bars_origin validation
+        # TODO: Better documentation in the code, of which values are allowed
+
+        cdef BarAggregation aggregation = self.bar_type.spec.aggregation
+
+        if type(self._time_bars_origin) is not pd.Timedelta and type(self._time_bars_origin) is not pd.DateOffset:
```

Comment:
This could also be validated in the constructor, the [Condition.type](https://github.com/nautechsystems/nautilus_trader/blob/develop/nautilus_trader/core/correctness.pyx#L151) function may be useful for standadization.